quilt/data.py

```diff
@@ -81,7 +82,12 @@ def _keys(self):
         """
         keys directly accessible on this object via getattr or .
         """
-        return self._store.keys(self._prefix)
+        group = self._store.get(self._prefix)
+        assert isinstance(group, dict), "{type} {grp}".format(type=type(group), grp=group)
+        assert TYPE_KEY in group
+        group.pop(TYPE_KEY)
```

Comment:
`del group[TYPE_KEY]`, since you don't care about the return value.

Don't even need the assert above since `del` and `pop` both raise an exception.